### PR TITLE
New custom analytics

### PR DESCRIPTION
**Pull request by matt-taylor934**
New custom analytics to report on:
- usage of the caffeinate binary on an interactive command line, due to a concern where a user may leave their machine awake and unlocked
- file sharing being enabled or disabled
- the root user being enabled or the password of an enabled root user being changed
- suspicious xattr activity where either all extended attributes are cleared or the com.apple.quarantine attribute cleared from a file

### CODE REVIEW

File: custom_analytic_detections/system_config_activity/file_sharing_config_changes [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# This Custom Analytic may be used to report on File Sharing (smbd) being enabled or disabled on the host endpoint.
+# This detection functions by monitoring for a launchctl command executed by macOS when a user toggles the File Sharing feature off or on inside System Preferences > Sharing.
+
+# Analytic Predicate(s):
+
+File Sharing Disabled
+$event.type == 1 AND $event.process.signingInfo.appid == "com.apple.xpc.launchctl" AND $event.process.args CONTAINS[cd] "unload" AND $event.process.args CONTAINS[cd] "/System/Library/LaunchDaemons/com.apple.smbd.plist"
+
+File Sharing Enabled
+$event.type == 1 AND $event.process.signingInfo.appid == "com.apple.xpc.launchctl" AND $event.process.args CONTAINS[cd] "load" AND $event.process.args CONTAINS[cd] "/System/Library/LaunchDaemons/com.apple.smbd.plist"
```
Comment (jbradley89) [inline custom_analytic_detections/system_config_activity/file_sharing_config_changes:12]: Same as above

**Review by jbradley89 — CHANGES_REQUESTED** [review]

File: custom_analytic_detections/system_config_activity/file_sharing_config_changes [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# file_sharing_config_changes
+#
+# This Custom Analytic may be used to report on File Sharing (smbd) being enabled or disabled on the host endpoint.
+# This detection functions by monitoring for a launchctl command executed by macOS when a user toggles the File Sharing feature off or on inside System Preferences > Sharing.
+
+# Analytic Predicate(s):
+
+File Sharing Disabled
+$event.type == 1 AND $event.process.signingInfo.appid == "com.apple.xpc.launchctl" AND $event.process.args CONTAINS[cd] "unload" AND $event.process.args CONTAINS[cd] "/System/Library/LaunchDaemons/com.apple.smbd.plist"
```
[the remark this answers is not in the document]
Comment (matt-taylor934) [reply custom_analytic_detections/system_config_activity/file_sharing_config_changes:9]: Just tested this and both do work as they're written in the commit, tested by turning FS off and on again a few times inside Sys Prefs.

Also tested your version for both and it also works so happy to swap to this if it's the preferred syntax.  Is there any reason we wouldn't add spaces before / after the plist path value?  Doing this caused it to fail whereas doing spaces just for 'unload' and 'load' worked fine.

File: custom_analytic_detections/system_config_activity/file_sharing_config_changes (hunk 1 above)
Comment (matt-taylor934) [reply custom_analytic_detections/system_config_activity/file_sharing_config_changes:12]: Replied above

**Comment by matt-taylor934** [comment]
Predicates updated as requested.

**Review by saljooki — APPROVED** [review]

**Review by stuartjash — APPROVED** [review]

**Review by jbradley89 — APPROVED** [review]

**Pull request merged**